components/FailedListItem.js

```diff
@@ -24,13 +38,52 @@ const styles = StyleSheet.create({
   },
 });
 
-const FailedListItem = ({ item, onPress }) => (
-  <TouchableOpacity style={styles.container} onPress={() => onPress(item.longitude, item.latitude)}>
-    <Text style={styles.text}>{I18n.t('locations/latitude', { value: item.latitude.toFixed(5) })}</Text>
-    <Text style={styles.text}>{I18n.t('locations/longitude', { value: item.longitude.toFixed(5) })}</Text>
-    <Text style={styles.error}>{item.error ? I18n.t(item.error) : I18n.t('locations/unknown_error') }</Text>
-  </TouchableOpacity>
-);
+const initialLatitudeDelta = 0.00200012;
+const initialLongitudeDelta = 0.00200006;
+const offset = 0.00010000;
+
+const FailedListItem = ({ item: { error, key, location: { latitude, longitude } }, onPress }) => {
+  const region = {
+    latitude: latitude + offset, // Keeps pin in view (Small map)
+    longitude,
+    latitudeDelta: initialLatitudeDelta,
+    longitudeDelta: initialLongitudeDelta,
+  };
+
+  return (
+    <TouchableOpacity style={styles.container} onPress={() => onPress(longitude, latitude)} key={key}>
+      <View style={styles.mapContainer}>
+        <MapView
+          style={styles.map}
+          region={region}
+          mapType="hybrid"
+          provider="google"
+          cacheEnabled
+          showsTraffic={false}
+          showsIndoors={false}
+          showsBuildings={false}
+          scrollEnabled={false}
+          zoomEnabled={false}
+          showsMyLocationButton={false}
+        >
+          <MapView.Marker
+            coordinate={{
+              latitude,
+              longitude,
+            }}
+            pinColor="red"
+            opacity={0.8}
+          />
+        </MapView>
+      </View>
+      <View style={styles.infoContainer}>
```

Comment:
I think we should probably highlight error message and show the longitude/latitude together at a smaller size.